chain/consensus/mir/state_manager.go

```diff
@@ -0,0 +1,263 @@
+package mir
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/multiformats/go-multiaddr"
+	"go4.org/sort"
+
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical"
+	"github.com/filecoin-project/mir/pkg/events"
+	"github.com/filecoin-project/mir/pkg/modules"
+	"github.com/filecoin-project/mir/pkg/pb/commonpb"
+	"github.com/filecoin-project/mir/pkg/pb/eventpb"
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+type Messages []byte
+
+type StateManager struct {
+
+	// The current epoch number.
+	currentEpoch t.EpochNr
+
+	// For each epoch number, stores the corresponding membership.
+	memberships    []map[t.NodeID]t.NodeAddress
+	membershipLock sync.Mutex
+
+	// Channel to send messages to Eudico.
+	ChainNotify chan []Messages
+
+	MirManager *Manager
+
+	reconfigurationVotes map[string]uint64
```

Comment:
The type is `uint64` but it's used only as `int` (converted to it). Can it directly be an `int`?

Reply:
fixed